# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1758972494 -7200
# Parent  6726be43deaaadb8264ab7b34a89c04c6b70dcb9
Bug 1956258 - Provide default account when needed.

This is https://phabricator.services.mozilla.com/D257098?id=1082030 which the reviewers rejected.
They preferred a solution that caused more regressions. Regressions ahoy!

diff --git a/mailnews/base/public/nsIMsgAccountManager.idl b/mailnews/base/public/nsIMsgAccountManager.idl
--- a/mailnews/base/public/nsIMsgAccountManager.idl
+++ b/mailnews/base/public/nsIMsgAccountManager.idl
@@ -31,8 +31,14 @@ interface nsIMsgAccountManager : nsISupp
    * @param aRemoveFiles  remove data directory (local directory) of this account
    */
   void removeAccount(in nsIMsgAccount aAccount, [optional] in boolean aRemoveFiles);
 
+
+  /**
+   * Find a new account that can serve as default.
+   */
+  void autosetDefaultAccount();
+
   /*
    * creates a new identity and assigns it a new, unique "key"
    */
   nsIMsgIdentity createIdentity();
diff --git a/mailnews/base/src/nsMsgAccountManager.h b/mailnews/base/src/nsMsgAccountManager.h
--- a/mailnews/base/src/nsMsgAccountManager.h
+++ b/mailnews/base/src/nsMsgAccountManager.h
@@ -154,13 +154,8 @@ class nsMsgAccountManager : public nsIMs
    * Check if the given account can be the set as the default account.
    */
   nsresult CheckDefaultAccount(nsIMsgAccount* aAccount, bool& aCanBeDefault);
 
-  /**
-   * Find a new account that can serve as default.
-   */
-  nsresult AutosetDefaultAccount();
-
   // sets the pref for the default server
   nsresult setDefaultAccountPref(nsIMsgAccount* aDefaultAccount);
 
   // Write out the accounts pref from the m_accounts list of accounts.
diff --git a/mailnews/compose/src/nsMsgComposeService.cpp b/mailnews/compose/src/nsMsgComposeService.cpp
--- a/mailnews/compose/src/nsMsgComposeService.cpp
+++ b/mailnews/compose/src/nsMsgComposeService.cpp
@@ -563,8 +563,11 @@ nsMsgComposeService::GetDefaultIdentity(
   nsCOMPtr<nsIMsgAccount> defaultAccount;
   rv = accountManager->GetDefaultAccount(getter_AddRefs(defaultAccount));
   NS_ENSURE_SUCCESS(rv, rv);
 
+  if (!defaultAccount) {
+    accountManager->AutosetDefaultAccount();
+  }
   return defaultAccount ? defaultAccount->GetDefaultIdentity(_retval) : NS_OK;
 }
 
 class nsMsgTemplateReplyHelper final : public nsIStreamListener,
